code/modules/mob/mob_defines.dm

```diff
@@ -186,3 +186,5 @@
 	var/list/additional_vision_handlers = list() //Basically a list of atoms from which additional vision data is retrieved
 
 	var/list/progressbars = null //for stacking do_after bars
+
+	var/datum/ai/ai						// Type abused. Define with path and will automagically create. Determines behaviour for clientless mobs.
```

Comment:
Can add a proc:
``` 
/mob/proc/get_ai_type()
    return ai
```
That way humans can override the proc via
```
/mob/living/human/get_ai_type()
    return species ? species.ai : ..()
    // Or: return species?.ai ?? ..()
    /* Or:
        if (species)
             return species.ai
        return ..()
    */
```
Reasoning in the next comment